src/main/java/io/cryostat/discovery/KubeApiDiscovery.java

```diff
@@ -188,8 +188,9 @@ KubernetesClient client() {
     }
 
     private boolean isCompatiblePort(EndpointPort port) {
-        return jmxPortNames.orElse(List.of()).contains(port.getName())
-                || jmxPortNumbers.orElse(List.of()).contains(port.getPort());
+        return (port.getName() != null && jmxPortNames.orElse(List.of()).contains(port.getName()))
+                || (port.getPort() != null
+                        && jmxPortNumbers.orElse(List.of()).contains(port.getPort()));
```

Comment:
```
$ jshell
|  Welcome to JShell -- Version 17.0.9
|  For an introduction type: /help intro

jshell> List.of().contains(null)
|  Exception java.lang.NullPointerException
|        at ImmutableCollections$ListN.indexOf (ImmutableCollections.java:716)
|        at ImmutableCollections$AbstractImmutableList.contains (ImmutableCollections.java:329)
|        at (#1:1)

jshell> new ArrayList<>().contains(null)
$2 ==> false
```

You could also simply do this replacement, maybe with something like `private static final List<String> EMPTY_PORT_NAMES = new ArrayList<>()`? I wonder if that might be a little clearer to read.

Reply:
Sounds good! How about now? I added one for port numbers too.